### PR TITLE
Templates, options, and services; Oh my!

### PR DESCRIPTION
**Pull request by colby**
* Options for setting which ip and port to listen on, as well as options for backend.
* Compiles to same version on all OSes.
* Installs all requires dependencies.
* Installs a service file into systemd. This breaks any Ubuntu 14.04 compatibility. I'll work on that.
* Generates a UUID secret just like a package installation.

UPDATED:
* Now installs init if systemd doesnt exist. Supports Ubuntu 12 and 14.
* Supports Varnish cache options.
* Supports most versions of CentOS/RHEL and Fedora.

WARNING:
* Init system is really basic/brittle. Using systemd is highly suggested. This only applies to CentOS 6, Ubuntu 12, and Ubuntu 14.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -21,7 +25,18 @@ service will be installed.
 name 'varnish'
 description 'this will install varnish'
 
-override_attributes()
+override_attributes(
+    ...
+    'varnish' => {
+        'ip'   => '192.168.0.666',
```
Comment (hardyjohnson) [inline README.md]: Should this be refactored into a separate key as well perhaps there are backend, frontend and admin keys which all have ip/port mappings?

File: templates/default/varnish.params.erb [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+# Admin interface listen address and port
+VARNISH_ADMIN_LISTEN_ADDRESS=<%= node['varnish']['admin']['ip'] %>
+VARNISH_ADMIN_LISTEN_PORT=<%= node['varnish']['admin']['port'] %>
+
+# Shared secret file for admin interface
+VARNISH_SECRET_FILE=/etc/varnish/secret
+
+# Backend storage specification, see Storage Types in the varnishd(5)
+# man page for details.
+VARNISH_STORAGE="file,/var/lib/varnish/varnish_storage.bin,1G"
```
Comment (hardyjohnson) [inline templates/default/varnish.params.erb]: The values in this file should probably all be exposed as attributes w/these values possibly as defaults.

File: README.md (hunk 1 above)
Comment (colby) [inline README.md]: There are, please see `attributes/default.rb`.

**Comment by colby** [comment]
I have a few more commits incoming.

**Comment by colby** [comment]
```bash
~/src/Work/COP/cookbooks/varnish / feature/templates_and_options
$ kli
Instance             Driver   Provisioner  Verifier  Transport  Last Action
default-ubuntu-1204  Vagrant  ChefZero     Busser    Ssh        Verified
default-ubuntu-1404  Vagrant  ChefZero     Busser    Ssh        Verified
default-ubuntu-1604  Vagrant  ChefZero     Busser    Ssh        Verified
default-centos-68    Vagrant  ChefZero     Busser    Ssh        Verified
default-centos-72    Vagrant  ChefZero     Busser    Ssh        Verified
default-fedora-22    Vagrant  ChefZero     Busser    Ssh        Verified
default-debian-711   Vagrant  ChefZero     Busser    Ssh        Verified
default-debian-86    Vagrant  ChefZero     Busser    Ssh        Verified
``` All providers/Operating Systems supported.

**Pull request merged**